client/client.go

```diff
@@ -94,20 +93,26 @@ func (c *Client) ListTasks(req *tes.ListTasksRequest) (*tes.ListTasksResponse, e
 }
 
 // CreateTask POSTs a Task message to /v1/tasks
-func (c *Client) CreateTask(msg []byte) (*tes.CreateTaskResponse, error) {
-	var err error
-	err = isValidTask(msg)
+func (c *Client) CreateTask(ctx context.Context, task *tes.Task) (*tes.CreateTaskResponse, error) {
+	verr := tes.Validate(task)
+	if verr != nil {
+		return nil, fmt.Errorf("invalid task message: %v", verr)
+	}
+
+	m := &jsonpb.Marshaler{}
+	msg, err := m.MarshalToString(task)
 	if err != nil {
-		return nil, fmt.Errorf("Not a valid Task message: %s", err)
+		return nil, fmt.Errorf("error marshaling task message: %v", err)
 	}
 
 	// Send request
-	r := bytes.NewReader(msg)
+	r := bytes.NewReader([]byte(msg))
```

Comment:
Is is possible to marshal directly to the request, or otherwise skip the step of marshaling to string?

Reply:
The data needs to be provided via `io.Reader` per https://golang.org/pkg/net/http/#NewRequest.  I can't think of how to get around this.

Reply:
Ya, I thought request was a writer, like response is, but it's not. Still, a small simplification:
```
var b bytes.Buffer
mar.Marshal(b, &task)
```